File: nm-space/projects/dost/src/utils/logger.ts
import { ILogger } from '../shares/log';
import { ipc } from './window';

class Logger implements ILogger {
  error(message: unknown, details?: Record<string, unknown>): void {
    console.error(...this.args(message, details));
    ipc.rendererLogger.error(message, details);
  }

  warn(message: unknown, details?: Record<string, unknown>): void {
    console.warn(...this.args(message, details));
    ipc.rendererLogger.warn(message, details);
  }

  info(message: unknown, details?: Record<string, unknown>): void {
    console.info(...this.args(message, details));
    ipc.rendererLogger.info(message, details);
  }

  debug(message: unknown, details?: Record<string, unknown>): void {
    console.debug(...this.args(message, details));
    ipc.rendererLogger.debug(message, details);
  }

  verbose(message: unknown, details?: Record<string, unknown>): void {
    console.debug(...this.args(message, details));
    ipc.rendererLogger.verbose(message, details);
  }

  private args(message: unknown, details?: Record<string, unknown>): unknown[] {
    const args = [message];
    if (details) {
      args.push(details);
    }
    return args;
  }
}

export const logger: ILogger = new Logger();

File: projects/android-device-agent/android/app/src/main/kotlin/com/dogu/deviceagent/Logger.kt
package com.dogu.deviceagent

import android.util.Log
import org.slf4j.LoggerFactory

interface ILogger {
    fun v(msg: String)
    fun i(msg: String)
    fun e(msg: String)
}

class Logger {
    companion object {
        private var deviceNickName = "";
        private val TAG = "deviceagent"
        private val logbackLogger = LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME)

        fun setDeviceNickName(nickName: String) {
            deviceNickName = nickName
        }
        fun formatMsg(msg: String): String {
            return "${Thread.currentThread().id}|$deviceNickName > $msg"

        }

        fun v(msg: String) {
            Log.v(TAG, formatMsg(msg))
            logbackLogger.debug(formatMsg(msg))
        }

        fun i(msg: String) {
            Log.i(TAG, formatMsg(msg))
            logbackLogger.info(formatMsg(msg))
        }

        fun w(msg: String) {
            Log.w(TAG, formatMsg(msg))
            logbackLogger.warn(formatMsg(msg))
        }

        fun e(msg: String) {
            Log.e(TAG, formatMsg(msg))
            logbackLogger.error(formatMsg(msg))
        }
    }
}

